api/handlers/v1/server.go

```diff
@@ -231,11 +312,41 @@ func resourceStatusToProto(status string) entropyv1beta1.Resource_Status {
 
 func resourceFromProto(res *entropyv1beta1.Resource) *domain.Resource {
 	return &domain.Resource{
-		Urn:     res.GetUrn(),
-		Name:    res.GetName(),
-		Parent:  res.GetParent(),
-		Kind:    res.GetKind(),
-		Configs: res.GetConfigs().GetStructValue().AsMap(),
-		Labels:  res.GetLabels(),
+		Urn:       res.GetUrn(),
+		Name:      res.GetName(),
+		Parent:    res.GetParent(),
+		Kind:      res.GetKind(),
+		Configs:   res.GetConfigs().GetStructValue().AsMap(),
+		Labels:    res.GetLabels(),
+		Providers: providerSelectorFromProto(res.GetProviders()),
+	}
+}
+
+func providerSelectorFromProto(ps []*entropyv1beta1.ProviderSelector) []domain.ProviderSelector {
+	var providerSelectors []domain.ProviderSelector
+
+	for _, p := range ps {
+		selector := domain.ProviderSelector{
+			Urn:    p.GetUrn(),
+			Target: p.GetTarget(),
+		}
+		providerSelectors = append(providerSelectors, selector)
+	}
+	return providerSelectors
+}
+
+func providerFromProto(pro *entropyv1beta1.Provider) *domain.Provider {
+	var conf map[string]interface{}
+	err := mapstructure.Decode(pro.GetConfigs(), &conf)
+	if err != nil {
+		return nil
```

Comment:
We don't really need to use mapstructure here. Proto go should be giving us an option to get a `map` from `Values` types.

Reply:
Oh yes, changing this.

Reply:
`pro.GetConfigs().GetStructValue().AsMap()` should work.